security/fma-withdrawals-root-in-block-header.md

```diff
@@ -0,0 +1,133 @@
+  - If there is an execution client bug, for example it is possible the root is (incorrectly) added to the header before the state is fully committed.
+
+  - If we were to ever introduce non empty `withdrawals` in the block body, this might override the mechanism introduced with this feature and invalidate the interpreation of the `withdrawalsRoot` field.
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  Temporary downtime for withdrawals, or loss of funds if not remediated in time to challenge a malicious proposal.
+
+  **Mitigations:**
+
+  - We rely on e2e tests to check for consistency between the outputs returned from op-node and those constructed manually in the old way.
+
+  - Instead of waiting for the failure mode to materialize and then writing a patch in a rush, we could add an optional config var to op-node to switch it back into the old behavior. Rolling out the fix would not then require any software releases.
+
+  - op-geth could be modified to log a critical error (triggering an alert) if the withdrawals list in the body is ever non empty.
+
+- **Detection:**
+  Fault proof monitoring systems may not detect this failure mode immediately, until an actor running patched software made a proposal or challenge.
+
+- **Recovery Path(s)**:
+  Fault proof infra would nee to be pointed at a patched op-node. The patch would restore the old behaviour for generating output roots.
+
+### FM2: Chain split from genesis due to divergent genesis block computation
+
+- **Description:**
+  The genesis state is, with the activation of Isthmus at genesis, now part of the genesis block. Therefore a consensus bug where clients compute a different genesis block from the same genesis state can cause a chain split.
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  **Mitigations:**
+  This should be caught by cross-client testing (kurtosis and devnets)
+
+- **Detection:**
+  If the bug made it to production, replica healthcheck alerts would fire as nodes diverged.
+
+- **Recovery Path(s)**:
+  The buggy client(s) would need to be patched to bring them back into compliance with the specification.
+
+### FM3: Failure of p2p network due to bug in new topic/message serde logic
+
+- **Description:**
+  Because this feature introduces a new p2p gossip topic and message serialization format, a bug can mean the failure of p2p gossip for any chain with Isthmus active. This would cause an unsafe chain halt.
+
+- **Risk Assessment:**
+
+  **High impact, low likelihood**
+
+  **Mitigations:**
+  We rely on [end-to-end testing](https://github.com/ethereum-optimism/optimism/blob/9249efc6343208f69283290fc9c5c8f6e7b243f8/op-service/eth/ssz_test.go#L251) (including fuzzing) to catch any bugs in this code path. We could run extended fuzzing campaigns.
+
+- **Detection:**
+  Continuous integration, or Kurtosis and/or devnet testing should catch this. Failing that, the bug makes it to production, our alerting infrastructure would notify us.
```

Comment:
This would lead to unsafe head stall, how we can make sure here that engineers can detect this issue?
with the error handling in the logs?

Reply:
We should route alerts from devnets somewhere that engineers see them.

Reply:
Added an action item for this! Please resolve this thread if you are happy @Ethnical